sklearn/ensemble/_hist_gradient_boosting/_binning.pyx

```diff
@@ -70,3 +74,67 @@ cdef void _map_num_col_to_bins(const X_DTYPE_C [:] data,
                 else:
                     left = middle + 1
             binned[i] = left
+
+
+def _map_cat_to_bins(const X_DTYPE_C [:, :] data,
+                     dict bin_categories,
+                     const unsigned char missing_values_bin_idx,
+                     X_BINNED_DTYPE_C [:, :] binned):
```

Comment:
kinda sad that we cannot enforce F-alignment because of the binning at prediction time...

Reply:
which is worth a comment in the description of `binned` below BTW